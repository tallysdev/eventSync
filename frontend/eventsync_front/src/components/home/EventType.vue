<template>
  <section class="d-block my-10">
    <v-row justify="center">
      <v-col cols="12" lg="10">
        <h2 class="text-h4 text-start mb-6 font-weight-bold">Escolha o seu tipo de evento</h2>
        <v-row class="d-flex justify-center">
          <v-col
            cols="auto"
            lg="2"
            v-for="type in eventTypes"
            :key="type.value"
            class="text-center"
            @click="redirectToEventList(type.value)"
            style="cursor: pointer"
          >
            <v-avatar size="100" color="primary">
              <v-icon size="50">{{ type.icon }}</v-icon>
            </v-avatar>
            <span class="d-block mt-6 text-h6">{{ type.title }}</span>
          </v-col>
        </v-row>
      </v-col>
    </v-row>
  </section>
</template>

<script setup lang="ts">
import { useRouter } from 'vue-router'

const router = useRouter()

const eventTypes = [
  { title: 'Todos', value: null, icon: 'mdi-calendar' },
  { title: 'Conferência', value: 'conference', icon: 'mdi-account-group' },
  { title: 'Oficina', value: 'workshop', icon: 'mdi-lightbulb-outline' },
  { title: 'Seminário', value: 'seminar', icon: 'mdi-school' },
  { title: 'Encontro', value: 'meetup', icon: 'mdi-handshake' }
]

const redirectToEventList = (type: string | null) => {
  router.push({ name: 'events', query: { type } })
}
</script>

<style scoped></style>
