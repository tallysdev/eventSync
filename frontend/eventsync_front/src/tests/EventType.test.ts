import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import EventType from '@/components/home/EventType.vue'

describe('EventType.vue', () => {
  const eventTypes = [
    { id: 1, name: 'Show', icon: 'mdi-music' },
    { id: 2, name: 'Esporte', icon: 'mdi-basketball' },
    { id: 3, name: 'Teatro', icon: 'mdi-drama-masks' },
    { id: 4, name: 'Cinema', icon: 'mdi-movie' }
  ]

  const wrapper = mount(EventType, {
    props: { eventTypes }
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the correct number of event types', () => {
    expect(wrapper.findAll('v-avatar')).toHaveLength(eventTypes.length)
  })
})
