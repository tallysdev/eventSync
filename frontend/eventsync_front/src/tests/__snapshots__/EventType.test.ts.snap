// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EventType.vue > renders correctly 1`] = `
"<section class="d-block my-10" eventtypes="[object Object],[object Object],[object Object],[object Object]">
  <v-row justify="center">
    <v-col cols="12" lg="10">
      <h2 class="text-h4 text-start mb-6 font-weight-bold">Escolha o seu tipo de evento</h2>
      <v-row class="d-flex justify-center">
        <v-col cols="auto" lg="2" class="text-center" style="cursor: pointer;">
          <v-avatar size="100" color="primary">
            <v-icon size="50">mdi-calendar</v-icon>
          </v-avatar><span class="d-block mt-6 text-h6">Todos</span>
        </v-col>
        <v-col cols="auto" lg="2" class="text-center" style="cursor: pointer;">
          <v-avatar size="100" color="primary">
            <v-icon size="50">mdi-account-group</v-icon>
          </v-avatar><span class="d-block mt-6 text-h6">Conferência</span>
        </v-col>
        <v-col cols="auto" lg="2" class="text-center" style="cursor: pointer;">
          <v-avatar size="100" color="primary">
            <v-icon size="50">mdi-lightbulb-outline</v-icon>
          </v-avatar><span class="d-block mt-6 text-h6">Oficina</span>
        </v-col>
        <v-col cols="auto" lg="2" class="text-center" style="cursor: pointer;">
          <v-avatar size="100" color="primary">
            <v-icon size="50">mdi-school</v-icon>
          </v-avatar><span class="d-block mt-6 text-h6">Seminário</span>
        </v-col>
        <v-col cols="auto" lg="2" class="text-center" style="cursor: pointer;">
          <v-avatar size="100" color="primary">
            <v-icon size="50">mdi-handshake</v-icon>
          </v-avatar><span class="d-block mt-6 text-h6">Encontro</span>
        </v-col>
      </v-row>
    </v-col>
  </v-row>
</section>"
`;
